integrationTests/multiShard/smartContract/executingSCCalls_test.go

```diff
@@ -0,0 +1,473 @@
+	)
+
+	err := proposerNodeShard1.txProcessor.ProcessTransaction(contractTx, generalRoundNumber)
+	assert.Nil(t, err)
+	_, err = proposerNodeShard1.accntState.Commit()
+	assert.Nil(t, err)
+
+	// Test that the gas for deploying the smart contract was substracted from the sender's account
+	acc, _ := proposerNodeShard1.node.GetAccount(hex.EncodeToString(senderAddressBytes))
+	expectedValue := big.NewInt(0)
+	expectedValue.Sub(mintingValue, big.NewInt(opGas*1))
+	assert.Equal(t, expectedValue, acc.Balance)
+	assert.Equal(t, senderNonce, acc.Nonce)
+
+	senderNonce++
+	generalRoundNumber++
+
+	// setting the sc deployment address (printed by the transaction processer)
+	scDeploymentAdddress, _ := hex.DecodeString("ca26d3e6152af91949295cc89f419413e08aa04ba2d5e1ed2b199b2ca8aabc2a")
+
+	// Now that the SC is deployed, we test a call from an account located in the second shard
+	addValue := uint64(100)
+	contractCallTx := createTx(
+		t,
+		secondShardAddressBytes,
+		scDeploymentAdddress,
+		receiverNonce,
+		big.NewInt(0),
+		1,
+		100,
+		"Add",
+		addValue,
+	)
+
+	// The second shard should process this tx as MoveBalance
+	err = proposerNodeShard2.txProcessor.ProcessTransaction(contractCallTx, generalRoundNumber)
+	assert.Nil(t, err)
+	_, err = proposerNodeShard2.accntState.Commit()
+	assert.Nil(t, err)
+
+	// Test again that the gas for calling the smart contract was substracted from the sender's account
+	acc, _ = proposerNodeShard2.node.GetAccount(hex.EncodeToString(secondShardAddressBytes))
+	// TODO: Afrer fees are implemented, from mintingValue we should substract gasLimit + fees until the other shard executes
+	//  the smart contract and a refund can be made with the remaining value the following rounds
+	assert.Equal(t, mintingValue, acc.Balance)
+	assert.Equal(t, receiverNonce, acc.Nonce)
+
+	receiverNonce++
+	generalRoundNumber++
+
+	// After second shard processed the transaction, tx should get into the first shard where the SC resides
+	err = proposerNodeShard1.txProcessor.ProcessTransaction(contractCallTx, generalRoundNumber)
+	assert.Nil(t, err)
+	_, err = proposerNodeShard1.accntState.Commit()
+	assert.Nil(t, err)
+
+	scAccount, _ := proposerNodeShard1.node.GetAccount(hex.EncodeToString(scDeploymentAdddress))
+	storedVal, _ := scAccount.DataTrieTracker().RetrieveValue([]byte("a"))
+	storedValBI := big.NewInt(0).SetBytes(storedVal)
+
+	assert.Equal(t, big.NewInt(int64(initialValueForInternalVariable + addValue)), storedValBI)
+}
+
+// Test within a network of two shards the following situation
+// 1. Node in first shard deploys a smart contract -> we also make sure that the resulting smart contract address falls within the same shard
+// 2. Another account within the second shard calls the smart contract, we make sure the smart contract is updated and the gas
+//  is substracted from the caller's balance at the end
+func TestProcessSCCallsInMultiShardArchitecture_FirstShardReceivesCallFromSecondShardWithSCResults(t *testing.T) {
```

Comment:
function too long, could you split it in 2 ?

Reply:
I extracted the code that could be reused, if still not ok I will try refactor more of it